src/libfuncs/cast.rs

```diff
@@ -218,7 +212,8 @@ pub fn build_downcast<'ctx, 'this>(
         }
         (Some(lower_check), None) => lower_check,
         (None, Some(upper_check)) => upper_check,
-        (None, None) => unreachable!(),
+        // its always in bounds since dst is larger than src (i.e no bounds checks needed)
+        (None, None) => entry.const_int(context, location, 1, 1)?,
```

Comment:
I think it would be better to check that before generating the boundary checks, this way we avoid generating unnecessary MLIR that will have to be optimized away.